Fix collada dae import for geometry with shared input offsets

Minon opened the pull request:
The Collada dae importer was not accounting for geometries that used inputs with shared offset values and assumed that there are always as many indices per facePoint as there were inputs.

This was causing the decoder to read/write vertices and indices to/from the wrong memory addresses, which can cause undefined behavior.

For example, if the geometry had the below inputs
```
<triangles material="material" count="3864">`
  <input semantic="VERTEX" source="#vertices" offset="0"/>
  <input semantic="NORMAL" source="#normals" offset="1"/>
  <input semantic="TEXCOORD" source="#map-0" offset="2" set="0"/>
  <input semantic="TEXCOORD" source="#map-1" offset="2" set="1"/>
  <p>287 0 0 288 1 1 286 2 2 0 3 3 ...
```
This means there are 3 index values for each face point, but the parser assumed there's 4 since there's 4 inputs.
The first triangle in this list would be pulling positions from index 287, 1, and 2 when it should be pulling from index 287, 288, and 286

This code change should help BC from silently crashing/creating garbled models when importing some dae files that would otherwise work in other tools.

Note: The dae files that ran into this issue were exported from blender 3.6.
This code has been tested and confirmed working with 2 dae files that were known to be working prior to this change.
It has **_not_** been tested against any files that contain vertex colors.

Minon committed:
Fix collada dae import for geometry with shared input offsets

The Collada dae importer was not accounting for geometries that used inputs with shared offset values and assumed that there are always as many indices per facePoint as there were inputs.

This was causing the decoder to read/write vertices and indices to/from the wrong memory addresses, which causes undefined behavior.

This should help prevent BC from crashing when importing some dae files that would otherwise work in other tools.

soopercool101 commented:
Looks good to me. I have a stable build I need to get released but will merge in once I get the last critical fix in for that, to allow some time for those on Canary to test properly

Minon committed:
Correcting Gimbal Lock fallback for Matrix decomposition

Noticed some bones were pointing the wrong way after importing some models.  The code that extracts rotations from Matrices apparently wasn't quite right in the very specific case where we have Gimbal Lock with the y rotation falling on -90 degrees.

I believe this is correct, going by page 4~5 of this paper https://eecs.qmul.ac.uk/~gslabaugh/publications/euler.pdf
